client/tinychain/ml/dnn.py

```diff
@@ -64,14 +193,29 @@ def train(self, inputs, cost):
                 A.append(A_l.copy())
                 Z.append(Z_l)
 
-            m = inputs.shape[0]
+            m = inputs.shape[0]   
             dA = cost(A[-1]).sum() / m
 
             updates = []
+            
+            initials=optimizer.create()
+            l_m,l_v,t=initials['l_m'],initials['l_v'],initials['t']
```

Comment:
It would be nice if we could define a standard interface for this. Right now there's no documentation to tell a developer how to make their own optimizer.

Reply:
from my perspective i just thought about 

- Initialize 
- Optimize
- Update

What do you think ?

Reply:
SGTM

Reply:
fine i will give some comments over there to help if someone cloned it